### ISSUE TITLE
Feature request: side Modals

### ISSUE THREAD

**Issue by MartijnCuppens** [twbs/bootstrap#27033]
I was recently working on a project where I themed a Bootstrap modal as a side drawer. I made an implementation that is compatible with the modal sizing classes. Maybe this is something that's nice to have in Bootstrap by default?

Demo: https://codepen.io/MartijnCuppens/pen/OwwjWV?editors=1100

![ezgif-2-21a3fc9652](https://user-images.githubusercontent.com/11559216/43793285-5b9277cc-9a7b-11e8-9a60-614215ee953b.gif)

PR incomming...

**Comment by MartijnCuppens** [comment]
@myorangeca, you can find the code in the PR https://github.com/twbs/bootstrap/pull/27035

**Comment by MartijnCuppens** [comment]
@coliff, I only searched for _"side modals"_ in the open issues and did not came across that PR. Thanks for mentioning it.

There are some differences in the PR's though:
- This version uses flexbox to push the `.modal-footer` to the bottom, not `position: absolute;`. This makes the footer optional
- This version is compatible with the modal sizes.

I see there is a bit of discussion in #21540 to make a new component instead of extending the modal functionality. The reason I chose to extend is that modals actually have all the functionality that the drawer need (the API, the keyboard accessibility). I think we would have a lot of duplicate code if we're going to make another component for this.
